File: leetcode-questions/java/src/test/java/com/leetcode/questions/insertinterval/InsertIntervalTest.java
package com.leetcode.questions.insertinterval;

import com.leetcode.questions.leetcode.editor.en.InsertInterval;
import com.leetcode.questions.support.GeneratedLeetCodeTestHarness;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import java.nio.file.Files;
import java.nio.file.Path;

public class InsertIntervalTest {

    private static final Path CASES_PATH = Path.of("generated-test-cases", "insert-interval.json");

    public InsertIntervalTest() {

    }

    @Test
    public void testCases() throws Exception {
        GeneratedLeetCodeTestHarness.run(InsertInterval.Solution.class, "insert", Files.readString(CASES_PATH));
    }
}

File: leetcode-questions/csharp/src/test/csharp/com/leetcode/questions/insertinterval/InsertIntervalTest.cs
using Xunit;
using System.Collections.Generic;
using com.leetcode.questions.leetcode.editor.en.insertinterval;
using com.leetcode.questions.support;

namespace com.leetcode.questions.insertinterval;

public class InsertIntervalTest {

    private const string CasesFileName = "insert-interval.json";

    public InsertIntervalTest() {

    }

    [Fact]
    public void TestCases() {
        GeneratedLeetCodeTestHarness.Run(typeof(Solution), "Insert", GeneratedLeetCodeTestHarness.ReadCasesFile(CasesFileName));
    }
}
